Makefile.toml

```diff
@@ -0,0 +1,321 @@
+env_scripts = [
+'''
+#!@duckscript
+crates = get_env CRATES
+crates = array %{crates}
+NAMES = set ""
+CRATE_NAMES = set ""
+YAMLS = set ""
+PATCHED_SVDS = set ""
+FORMATTED_SVDS = set ""
+RUST_DIRS = set ""
+for crate in ${crates}
+    crate_yamls = glob_array devices/${crate}*.yaml
+    for yaml in ${crate_yamls}
+        serie = substring ${yaml} 8
+        serie = substring ${serie} -5
+        NAMES = concat ${NAMES} " " ${serie}
+        CRATE_NAMES = concat ${CRATE_NAMES} " " ${crate}
+        YAMLS = concat ${YAMLS} " " ${yaml}
+        PATCHED_SVDS = concat ${PATCHED_SVDS} " " svd/${serie}.svd.patched
+        FORMATTED_SVDS = concat ${FORMATTED_SVDS} " " svd/${serie}.svd.formatted
+        RUST_DIRS = concat ${RUST_DIRS} " " ${crate}/src/${serie}/
+    end
+    release ${crate_yamls}
+end
+NAMES = trim ${NAMES}
+set_env NAMES ${NAMES}
+CRATE_NAMES = trim ${CRATE_NAMES}
```

Comment:
Why do you need trim here? Is it possible some vars will be empty in the loop above?